packages/blade/src/utils/useBreakpoint/useBreakpoint.ts

```diff
@@ -0,0 +1,144 @@
+
+type BreakpointAndDevice = {
+  matchedBreakpoint: Breakpoint;
+  matchedDeviceType: DeviceType;
+};
+
+const useBreakpoint = ({ breakpoints }: { breakpoints: Breakpoints }): BreakpointAndDevice => {
+  const supportsMatchMedia =
+    typeof document !== 'undefined' && typeof window?.matchMedia === 'function';
+
+  const breakpointsTokenAndQueryCollection = useMemo(
+    () =>
+      (supportsMatchMedia
+        ? Object.entries(breakpoints).map(
+            ([breakpointTokenName, breakpointSize], index, breakpointsArray) => {
+              let mediaQuery = '';
+
+              if (breakpointTokenName === 'max') {
+                mediaQuery = `screen and (min-width: ${breakpointSize}px)`;
+              } else if (breakpointsArray[index - 1]) {
+                mediaQuery = `screen and (min-width: ${
+                  breakpointsArray[index - 1][1] + 1
+                }px) and (max-width: ${breakpointSize}px)`;
+              } else {
+                mediaQuery = `screen and (max-width: ${breakpointSize}px)`;
+              }
+
+              return [breakpointTokenName, breakpointSize, mediaQuery];
+            },
+          )
+        : []) as [keyof Breakpoints, number, string][],
+    [breakpoints, supportsMatchMedia],
+  );
+
+  const getMatchedDeviceType = useCallback((matchedBreakpoint: Breakpoint): DeviceType => {
+    let matchedDeviceType: DeviceType = 'mobile';
+    if (typeof navigator !== 'undefined' && navigator.product === 'ReactNative') {
+      // react-native
+      matchedDeviceType = 'mobile';
+    } else if (typeof document !== 'undefined') {
+      // browser
+      if (matchedBreakpoint && ['xs', 's', 'm'].includes(matchedBreakpoint)) {
+        // tablet is also categorised as mobile
+        matchedDeviceType = 'mobile';
+      } else {
+        matchedDeviceType = 'desktop';
+      }
+    } else if (typeof process !== 'undefined') {
+      // node
+      //@TODO: Check for useragent for node
+      matchedDeviceType = 'desktop';
+    }
+    return matchedDeviceType;
+  }, []);
+
+  const getMatchedBreakpoint = useCallback(
+    (event?: MediaQueryListEvent): Breakpoint => {
+      const matchedBreakpoint =
+        breakpointsTokenAndQueryCollection.find(([_, __, mediaQuery]) => {
+          if (event?.media === mediaQuery) {
+            return true;
+          }
+          if (window.matchMedia(mediaQuery).matches) {
+            return true;
+          }
+          return false;
+        })?.[0] ?? undefined;
+
+      return matchedBreakpoint;
+    },
+    [breakpointsTokenAndQueryCollection],
+  );
+
+  const [breakpointAndDevice, setBreakpointAndDevice] = useState(() => {
+    const matchedBreakpoint = getMatchedBreakpoint();
+    const matchedDeviceType = getMatchedDeviceType(matchedBreakpoint);
+    return {
+      matchedBreakpoint,
+      matchedDeviceType,
+    };
+  });
+
+  useEffect(() => {
+    if (!supportsMatchMedia) {
+      return undefined;
+    }
```

Comment:
we basically return early here when the supports matchmedia doesn't exist. this will be true for ssr and react-native apps

Reply:
it's self-explanatory no. what shall I add? yaha pe I've added so that it's easier for PR reviewer